### PR TITLE
Use the latest LibP2P

### PR DESCRIPTION
**Pull request by zah**
Depends on https://github.com/status-im/nim-libp2p/pull/204

### CODE REVIEW

File: beacon_chain/conf.nim [hunk 1]
```diff
@@ -78,7 +78,7 @@ type
       libp2pAddress* {.
         defaultValue: defaultListenAddress(config)
         desc: "Listening address for the Ethereum LibP2P traffic."
-        name: "listen-address"}: IpAddress
+        name: "listen-address" }: ValidIpAddress
```
Comment (arnetheduck) [inline beacon_chain/conf.nim:81]: how's the `ValidIpAddress` different?

Comment (zah) [reply beacon_chain/conf.nim:81]: It's supposed to lack invalid states (it must always be initialized with `ipv4(...)`, `ipv6(...)` or a string-parsing function)

Comment (arnetheduck) [reply beacon_chain/conf.nim:81]: doh - `echo default(ValidIpAddress)` prints an empty ipv6 though

Comment (zah) [reply beacon_chain/conf.nim:81]: No, this shouldn't be possible. I had forgotten to push a local modification in stew/shims/net.

**Pull request merged**